backend/src/src-bdpimport/src/main/java/com/bosch/osmi/sw360/bdp/datasink/thrift/ThriftUploader.java

```diff
@@ -0,0 +1,209 @@
+/*
+ * Copyright (c) Bosch Software Innovations GmbH 2015.
+ * Part of the SW360 Portal Project.
+ *
+ * This program is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU General Public License Version 2.0 as published by the
+ * Free Software Foundation with classpath exception.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU General Public License version 2.0 for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License along with
+ * this program (please see the COPYING file); if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
+ */
+package com.bosch.osmi.sw360.bdp.datasink.thrift;
+
+import com.bosch.osmi.bdp.access.api.model.ProjectInfo;
+import com.bosch.osmi.sw360.bdp.datasource.BdpApiAccessWrapper;
+import com.bosch.osmi.sw360.bdp.entitytranslation.BdpComponentToSw360ComponentTranslator;
+import com.bosch.osmi.sw360.bdp.entitytranslation.BdpComponentToSw360ReleaseTranslator;
+import com.bosch.osmi.sw360.bdp.entitytranslation.BdpLicenseToSw360LicenseTranslator;
+import com.bosch.osmi.sw360.bdp.entitytranslation.BdpProjectInfoToSw360ProjectTranslator;
+import com.siemens.sw360.datahandler.thrift.RequestStatus;
+import com.siemens.sw360.datahandler.thrift.components.Component;
+import com.siemens.sw360.datahandler.thrift.components.Release;
+import com.siemens.sw360.datahandler.thrift.licenses.License;
+import com.siemens.sw360.datahandler.thrift.projects.Project;
+import com.siemens.sw360.datahandler.thrift.users.User;
+import com.siemens.sw360.datahandler.thrift.bdpimportstatus.BdpImportStatus;
+import org.apache.log4j.Logger;
+import org.apache.thrift.TException;
+
+import java.util.*;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+// TODO: this contains logic and knowledge of the inner relations => move them to entitytranslation?
+public class ThriftUploader {
```

Comment:
I somehow don't find it back on github and I don't see from below what we
could add. Is it still relevant?

-- 
Dr. Birgit Heydenreich * birgit.heydenreich@tngtech.com * +49-174-3410216
TNG Technology Consulting GmbH, Betastr. 13a, 85774 Unterföhring
Geschäftsführer: Henrik Klagges, Christoph Stock, Dr. Robert Dahlke
Sitz: Unterföhring * Amtsgericht München * HRB 135082

Am 18.05.2016 um 10:10 schrieb Alex Borodin:
> In
> backend/src/src-bdpimport/src/main/java/com/bosch/osmi/sw360/bdp/datasink/thrift/ThriftUploader.java
> <https://github.com/sw360/sw360portal/pull/167#discussion_r63661829>:
> 
>> +import com.siemens.sw360.datahandler.thrift.RequestStatus;
>> +import com.siemens.sw360.datahandler.thrift.components.Component;
>> +import com.siemens.sw360.datahandler.thrift.components.Release;
>> +import com.siemens.sw360.datahandler.thrift.licenses.License;
>> +import com.siemens.sw360.datahandler.thrift.projects.Project;
>> +import com.siemens.sw360.datahandler.thrift.users.User;
>> +import com.siemens.sw360.datahandler.thrift.bdpimportstatus.BdpImportStatus;
>> +import org.apache.log4j.Logger;
>> +import org.apache.thrift.TException;
>> +
>> +import java.util.*;
>> +import java.util.function.Function;
>> +import java.util.stream.Collectors;
>> +
>> +// TODO: this contains logic and knowledge of the inner relations => move them to entitytranslation?
>> +public class ThriftUploader {
> 
> I thought you may want to add it. If you don't - that's totally fine with me.
> 
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub
> <https://github.com/sw360/sw360portal/pull/167/files/435cce4a2f88228b253fe2eceaba26ae26cb93d1..73f055c4bed8a708cdf4a0246255089d1cc478fc#r63661829>
>